### ISSUE TITLE
Basic Bot Not Authenticating No Matter What Password I Use.

### ISSUE THREAD

**Issue by hackdefendr** [skyware-js/bot#22]
Distributor ID:	Debian
Description:	Debian GNU/Linux 12 (bookworm)
Release:	12
Codename:	bookworm
NodeJS: 22 (LTS)
SkywareJS/Bot: v0.38

I have tried every single method of authenticating to BlueSky with the most basic code. Nothing I do is working.

```
import { Bot } from '@skyware/bot';

async function createBot() {
  const bot = new Bot();
  await bot.login({
    identifier: process.env.username,
    password: process.env,password,
    service: process.env.service
  });
  bot.post('Hello, Bluesky! This is a test message.');
  bot.logout();
}

createBot();
```
I have tried the password created via pdsadmin, I have tried an app password, and I have tried an invite-code. I have used goat to validate my bots account. I can login to my bot account via the web and from other clients. But not from Skyware/Bot, this method is the only method that doesn't work.
```
file:///root/skybot/node_modules/@skyware/bot/dist/bot/Bot.js:98
            throw new Error("Failed to log in — double check your credentials and try again.", {
                  ^
Error: Failed to log in — double check your credentials and try again.
    at file:///root/skybot/node_modules/@skyware/bot/dist/bot/Bot.js:98:19
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
    at async Bot.login (file:///root/skybot/node_modules/@skyware/bot/dist/bot/Bot.js:97:26)
    at async createBot (file:///root/skybot/bot.js:12:3) {
  [cause]: XRPCError: AuthenticationRequired > Invalid identifier or password
      at XRPC.request (file:///root/skybot/node_modules/@atcute/client/dist/rpc.js:82:19)
      at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
      at async CredentialManager.login (file:///root/skybot/node_modules/@atcute/client/dist/credential-manager.js:142:21)
      at async Bot.login (file:///root/skybot/node_modules/@skyware/bot/dist/bot/Bot.js:97:26)
      at async createBot (file:///root/skybot/bot.js:12:3) {
    status: 401,
    kind: 'AuthenticationRequired',
    description: 'Invalid identifier or password',
```
Would someone please tell me what I am doing wrong? The code should Login, Post, and Logout.

Please help.

**Comment by UmbyUmbreon** [comment]
Looks like a typo, you've got `process.env,password` when you should be using `process.env.password`.

**Comment by hackdefendr** [comment]
That typo is not the problem, it only exists in this issue.  I even shorten the code, putting my credentials directly in the file and it still doesn't work.

```
import { Bot } from '@skyware/bot';

const bot = new Bot();
await bot.login({
  identifier: 'btbot.hackdefendr.com',
  password: '<password>',
  service: 'https://hackdefendr.com'
});

await bot.post('Hello, Bluesky! This is a test message.');
bot.logout();
```

**Comment by hackdefendr** [comment]
Maybe we should start from the beginning. Which password is expected (regular?, app?, invite?) and is `password` the expected variable name? I am down to just trying to login now. If I use tools like pdsadmin, bsky-cli, or goat, I am able to login. If I switch accounts in the web UI for bsky.app, I am able to login. Only from this code does not work and I am not getting answers from the logs.

**Comment by moll-dev** [comment]
@hackdefendr I found a fix, it should look like this:
```
const bot = new Bot({service: "<service URL>"});
await bot.login({
	identifier: "<labeler DID>",
	password:"<password>",
});
```

**Comment by hackdefendr** [comment]
Awesome. That worked.

Note: I set up a bot using the atproto library as a comparison and that's why I raised this issue. Because I can auth just fine with the atproto library code.

**Comment by moll-dev** [comment]
I want to say there's some kind of mechanism to look up a PLC's service automatically and maybe skyware doesn't do this? I don't remember off the top of my head.

**Comment by futurGH** [comment]
You coulddd but I don't think any other library does that, + don't really see the value when presumably you should know what PDS your account is on.

**Comment by hackdefendr** [comment]
So what you are saying is that we should just assume what the Skyware code is expecting for particular value? Maybe we should just automatically know that `identifier` means DID with Skyware, but with ATPROTO `identifier` means `username` - I know this because they documented it. Ah yes, the documentation, maybe that's what Skyware needs? Better documentation.

**Comment by futurGH** [comment]
It can be DID, username, or password — it's up to your PDS to decide what it accepts. You were passing `service` into the wrong method entirely.

**Comment by hackdefendr** [comment]
Show me the documentation that details this? It doesn't exist. And I tried every possible method except the DID for username, because it isn't documented to do it that way. Where as ATPROTO tells you right there in plain English what is expected in a field.

**Comment by moll-dev** [comment]
It's in the docs: https://skyware.js.org/guides/bot/introduction/getting-started/. I just grepped for "service" in the code base and it led me to `[BotPostOptions](https://skyware.js.org/docs/bot/types/BotPostOptions)`
